src/WebExploitation/SQLInjections/Blind SQL Injection.md

```diff
@@ -0,0 +1,95 @@
+
+## Table of Contents
+
+- [SQL Injection: Blind SQL Injection](#sql-injection-blind-sql-injection)
+  - [Introduction](#introduction)
+  - [Table of Contents](#table-of-contents)
+  - [How to identify](#how-to-identify)
+  - [Code/tools/websites](#codetoolswebsites)
+    - [Tools](#tools)
+    - [Websites](#websites)
+  - [Sample problem](#sample-problem)
+    - [Problem Statement](#problem-statement)
+    - [Solution](#solution)
+  - [References](#references)
+  - [Conclusion](#conclusion)
+
+## How to identify
+
+To identify Blind SQL Injection vulnerabilities, pay attention to subtle differences in application responses. Indicators include:
+
+1. **Behavior Testing:** Entering SQL syntax that does not produce visible errors but causes changes in application behavior, such as differences in response times or content.
+2. **Boolean-based Testing:** Using boolean conditions (true/false) to determine the existence of a vulnerability.
+3. **Time-based Testing:** Using time delays (e.g., `SLEEP` function) to measure response times and infer database information. (More on this in [here](./time_based_blind_injections.md))
+
+Example:
+```plaintext
+http://example.com/item.php?id=1 AND 1=1 -- (Normal response)
+http://example.com/item.php?id=1 AND 1=2 -- (Different response)
+```
+ or maybe a form
+ ```
+ Product ID: A97HY  (Normal response)
+ Product ID: true --  (Different response)
+ ```
+
+## Code/tools/websites
+
+### Tools
+- **SQLMap:** An automated tool for SQL injection and database takeover.
+- **Burp Suite:** A web vulnerability scanner with tools for manual testing.
+- **Havij:** An automated SQL injection tool.
+
+### Websites
+
+> https://ctflearn.com/challenge/430 [Quite hard]
+> 
+Hint: First use auth by-pass then start applying blind SQL.
+## Sample problem
+
+### Problem Statement
+This is based on a real CTF question, from the RaziCTF 2020.
+In this question you are given a login page and a `/login` endpoint we can post credentials to. When trying to log in it just says `You are not allowed to login` which seems quite odd. 
+```
+http://130.185.122.155:8080/login 
+```
+Identify if the application is vulnerable to Blind SQL Injection.
+After this you need to login and get the length of the database.
+
+### Solution
+1. Test for SQL injection vulnerability using boolean-based testing:
+   ```sql
+   http://130.185.122.155:8080/login AND 1=1 -- (Normal response)
+   http://130.185.122.155:8080/login AND 1=2 -- (Different response)
+   ```
+
+
+2. Use what we learnt in authentication by-pass to login.
+   ```SQL
+   http://130.185.122.155:8080/login " ' or 1--"
```

Comment:
Again, can you double-check this?

Reply:
that was meant to be a request param, login?data=...